crates/router/tests/connectors/utils.rs

```diff
@@ -53,7 +58,12 @@ pub trait ConnectorActions: Connector {
         let state =
             routes::AppState::with_storage(Settings::new().unwrap(), StorageImpl::PostgresqlTest)
                 .await;
-        integration.execute_pretasks(&mut request, &state).await?;
+
+        let session_token = request
+            .get_session_token(&state, &connector)
+            .await
+            .expect("Session token call failed");
+        request.request.session_token = session_token.map(|s| s.session_token);
```

Comment:
Why is the session token being set separately instead of within `get_session_token()` method? Is it possible to take a mutable borrow?